src/historyRoutes.js

```diff
@@ -0,0 +1,51 @@
+import express from 'express';
+const router = express.Router()
+import authMiddleware from './auth/authMiddleware.js';
+import { pgPool } from './db/dbClient.js';
+
+// NOTE: all these routes are prefixed with /api/history (see server.js)
+
+/**
+ * @api {get} /api/history Get history
+ * @apiPermission auth
+ * @apiGroup history
+ *
+ * @apiSuccess {Object[]} Trades array of trade item objects pertaining to user
+ * @apiSuccessExample {json} Success-Response:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "Trades": [
+ *         {
+ *           "trade_id": "51cb778c-6129-443e-8be4-0b8a96fa808a",
+ *           "date_executed": "2021-02-12",
+ *           "user_id": "61f004d1-56c0-4303-af89-fe39e904cbce",
+ *           "ticker": "MSFT",
+ *           "action": "buy",
+ *           "quantity": "300",
+ *           "unit_price_executed": "140.23",
+ *         },
+ *         {
+ *           "trade_id": "51cb778c-6129-443e-8be4-0b8a96fa808b",
+ *           "date_executed": "2021-02-12",
+ *           "user_id": "61f004d1-56c0-4303-af89-fe39e904cbce",
+ *           "ticker": "MSFT",
+ *           "action": "sell",
+ *           "quantity": "300",
+ *           "unit_price_executed": "144.23",
+ *         }
+ *       ]
+ *     }
+ * @apiUse auth
+ */
+
+
+router.get('/', authMiddleware, async (req, res) => {
+  const { rows }  = await pgPool.query(`
+    SELECT * FROM Trades WHERE user_id = $1;`, 
+    [ req.user.user_id ]);
+  return res.json({
+    Trades: rows,
+  });
+});
+
```

Comment:
I believe you need to implement pagination. For now, using LIMIT and OFFSET should be okay.

Reply:
make sure to specify this in the query string params

Reply:
Also, you need to order by date (ascending) since we want to get history of trades chronologically.

Reply:
Added the ORDER BY ASC;
Lorance said pagination part can be done in the front end.

Reply:
@SeunghyunDKim  @lorance19 it's really not a good practice & frowned upon to do pagination in the frontend, but given the scope of this project we can allow it for now.